uwp/uwp_main.cpp

```diff
@@ -586,6 +613,45 @@ void App::OnPackageInstalling(PackageCatalog^ sender, PackageInstallingEventArgs
 	}
 }
 
+void App::ParseProtocolArgs(Windows::ApplicationModel::Activation::IActivatedEventArgs^ args, int *argc, std::vector<char*> *argv, std::vector<std::string> *argvTmp)
+{
+	argvTmp->clear();
+	argv->clear();
+
+	if (args->Kind == ActivationKind::Protocol)
+	{
+		ProtocolActivatedEventArgs^ protocolArgs = dynamic_cast<Windows::ApplicationModel::Activation::ProtocolActivatedEventArgs^>(args);
+		Windows::Foundation::WwwFormUrlDecoder^ query = protocolArgs->Uri->QueryParsed;
+
+		try
+		{
+			//protocol activation is in format 0=argValue0&1=argValue1...
+			//where argument name is a number from 0 to 9
+			for (int i = 0; i < 10; i++)
```

Comment:
could be improved by implementing support for standard html encoded characters rather than having to have each space be a new arg

Reply:
I'm not sure I understand what needs to be done. Currently, the idea is to pass arguments using URI scheme and convert those into argv array. When handling URI activation, parameters are passed as named parameters.
https://docs.microsoft.com/en-us/windows/uwp/launch-resume/handle-uri-activation
and we're using GetFirstValueByName to find parameters and convert them to each argv.
I guess we could use something like: retroarch:?0=retroarch&command=-L cores\stella_libretro.dll <full_path_to_rom>"
but then we additionally need to implement parsing of that one command into argv array. It seems excessive to do so, when we already have methods provided by UWP API.
Let me know if there's a better way to do the implementation.